src/renderers/shaders/ShaderChunk/lights_fragment_begin.glsl.js

```diff
@@ -20,14 +20,10 @@ geometry.position = - vViewPosition;
 geometry.normal = normal;
 geometry.viewDir = normalize( vViewPosition );
 
-#ifdef USE_CLEARCOAT_NORMALMAP
+#ifdef PHYSICAL
 
 	geometry.clearCoatNormal = clearCoatNormal;
 
-#else
-
-	geometry.clearCoatNormal = geometryNormal;
```

Comment:
Hmm, was this change intentional?

Reply:
Yes, `clearCoatNormal` `variable` is every declared if the material is `PHYSICAL`, now this is independent of has defined a `material.clearCoatNormalMap`. This too fix the design proposed in `lights_fragment_begin` of not having any native MAPS, only native LIGHTS.

Reply:
Hmm, I'll have to study this closely tomorrow but. I added this check so `material.normalMap` doesn't affect clear coat when `material.clearCoatNormalMap` is not defined.

That's how the top-right material works here:
https://raw.githack.com/mrdoob/three.js/dev/examples/webgl_materials_clearcoat_normalmap.html

Does the change still support hat behavior?

Reply:
Hmm, looking at the code a bit closely... yeah, I think it does.

Reply:
I guess we can also do this:

```glsl
	#ifdef PHYSICAL
	vec3 clearCoatNormal;
	#endif
```

https://github.com/mrdoob/three.js/blob/dev/src/renderers/shaders/ShaderChunk/common.glsl.js#L42

Reply:
Yeah,  `geometryNormal = normal` is seted before of `NormalMap`, in this sense `geometryNormal` is a pure normal. 

After `clearCoatNormal = geometryNormal` is seted. 

`normal` can be a variation of `material.normalMap` and `clearCoatNormal` a variation of `material.clearCoatNormalMap`. 

`normalMap` and `clearCoatNormalMap` both dont not depend on each other. Both in theory is derived from `geometryNormal`.

Reply:
Added, I too made this change `ifnref STANDARD` to `ifdef PHYSICAL`, I think that this is more readable https://github.com/mrdoob/three.js/pull/17256/commits/b244ee366dc865019946d2be246d9d2d6f871534

Reply:
With relation of `NormalNode.NORMAL`, now this use `geometryNormal`, in this sense, `NormalNode.NORMAL` is not affected with changes of `normal` or `clearCoatNormal`.

Reply:
> Added, I too made this change `ifndef STANDARD` to `ifdef PHYSICAL`, I think that this is more readable [b244ee3](https://github.com/mrdoob/three.js/commit/b244ee366dc865019946d2be246d9d2d6f871534)

I think this is great, but other defines would have to be changed.

I've been confused by this myself. It seems like in some places, the code assumes `PHYSICAL` is the inverse of `STANDARD` (e.g.: _lights_fragment_maps.glsl.js_), while in other places it's assume `STANDARD` implies `PHYSICAL` (e.g.: _envmap_pars_fragment.glsl.js_).

@DanielSturk and I have been talking about making the defines more expressive. This mix-up would be covered in that PR. What are your opinions?

Reply:
I think, make this system more modular and it can help performance in customized materials like NodeMaterial for example, instead of having only two materials, a full (PHYSICAL) and another half (STANDARD) add intermediates modules in `STANDARD` that are activated in `PHYSICAL` declaration, for example:

```javascript
#ifdef PHYSICAL
	#define CLEAR_COAT
	#define SHEEN
	#define TRANSPARENCY
#endif
```

Reply:
> I think, make this system more modular and it can help performance in customized materials like NodeMaterial

I agree, this is what we were looking at too.